wp-content/themes/core/components/container/Container.php

```diff
@@ -8,33 +8,35 @@
  * A component to contain arbitrary html and take classes/attributes. Useful when
  * composing more complex ui's in other controllers
  *
- * @property string   $content
+ * @property string   $content_component
+ * @property array    $content
  * @property string   $tag
  * @property string[] $classes
  * @property string[] $attrs
  */
-class Container extends Context {
-	public const CONTENT = 'content';
-	public const TAG     = 'tag';
-	public const CLASSES = 'classes';
-	public const ATTRS   = 'attrs';
+class Container extends Component {
 
-	protected $path = __DIR__ . '/container.twig';
+	public const CONTENT_COMPONENT = 'content_component';
+	public const CONTENT           = 'content';
+	public const TAG               = 'tag';
+	public const CLASSES           = 'classes';
+	public const ATTRS             = 'attrs';
 
-	protected $properties = [
-		self::CONTENT => [
-			self::DEFAULT => '',
-		],
-		self::TAG     => [
-			self::DEFAULT => 'div',
-		],
-		self::CLASSES => [
-			self::DEFAULT       => [ 'c-container' ],
-			self::MERGE_CLASSES => [],
-		],
-		self::ATTRS   => [
-			self::DEFAULT          => [],
-			self::MERGE_ATTRIBUTES => [],
-		],
-	];
+	protected function defaults(): array {
+		return [
+			self::CONTENT_COMPONENT => 'content-block/Content_Block.php',
+			self::CONTENT           => [],
+			self::TAG               => 'div',
+			self::CLASSES           => [ 'c-container' ],
+			self::ATTRS             => [],
+		];
+	}
+
+	public function render(): void {
+		?>
+		<{{ tag }} {{ classes|stringify }} {{ attrs|stringify }}>
+			{{ component( content_component, content ) }}
```

Comment:
Why better than a forced component usage pattern (see my note on the the Content_Block component below), it feels strange to me to define a "default" component (or really any component at all) at this level. It feels contrary to the idea that the components are dumb templates, rendering whatever data is passed to them without any opinion.

Reply:
Great question! 

So, in this case, it's because what gets output here needs to be variable by nature, even within a single project. It's a decision that needs to get made at a higher level than this component every time, so we can't have a specific one hardcoded into this Component itself. 

As to the default, I could definitely see an argument being made to have it default to blank, and then maybe just wrap the output in a check for that value being defined. A default helps make it a little less "wait, why is nothing showing?", but I think your point is definitely valid and I wouldn't be against making a move towards that.